feat: DIA-2075: [FE] Homepage projects list perf optimization

nick-skriabin opened the pull request:
Moving some components and utilities to `core` and `ui` to avoid using `@humansignal/shad` directly.

This pull request includes several changes to the `web/libs` directory aimed at enhancing the utility functions and re-exporting components for better modularity and reusability. The most important changes are as follows:

### Enhancements to utility functions:

* [`web/libs/core/src/lib/utils/helpers.ts`](diffhunk://#diff-c1fdc3ece36c32afd334cf9baf40d4b858d62fc76ad75322d66798abc7873670R31-R32): Exported the `cn` function from `@humansignal/ui/shad/utils/index` to facilitate smart merging of class names.
* [`web/libs/ui/src/shad/utils/index.ts`](diffhunk://#diff-01ed459c92d9de537f51611e9a060a2e59b944988a32ff043d4bd65b5b342e27R4-R6): Added a comment to document the `cn` function, which smartly merges class names using `clsx` and `twMerge`.

### Re-exporting components for modularity:

* [`web/libs/ui/src/index.ts`](diffhunk://#diff-1b7b120e6194326932be999557506481a8f7991798c7c3ee3d6e4edc42ad41a4R16-R17): Re-exported all components from the `shadcn` module to make them available for use in other parts of the application.
* [`web/libs/ui/src/shadcn.ts`](diffhunk://#diff-58d4051963c17002d98a7398a27dd21038be599dc818e6b300fc230810d9f4c3R1-R3): Added re-exports for `Badge`, `BadgeProps`, and `badgeVariants` from `./shad/components/ui/badge` to provide raw shadcn components for reuse.

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Re-export `cn` and `Badge`

nick-skriabin committed:
Correctly export type

nick-skriabin committed:
Migrate flag

nick-skriabin committed:
Skeleton loader

nick-skriabin committed:
Merge branch 'develop' into fb-dia-2075/homepage-perf

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14166043743)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-2075/homepage-perf'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14166043743

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14166043743

nick-skriabin committed:
Fix class name computation
